### PR TITLE
Convert default ALQ type to GRAT when gaslift optimization is active

### PR DESCRIPTION
**Pull request by joakim-hove**
A possible alternative to: #2966 

Downstream: https://github.com/OPM/opm-simulators/pull/3831

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/input/eclipse/Schedule/Schedule.hpp [hunk 1]
```diff
@@ -87,6 +87,7 @@ namespace Opm
         std::optional<int> output_interval;
         double sumthin{-1.0};
         bool rptonly{false};
+        bool gaslift_opt{false};
```
Comment (bska) [inline opm/input/eclipse/Schedule/Schedule.hpp:90]: If we're going to use a `bool` can we please name it something more descriptive than `gaslift_opt`&mdash;e.g., `alq_is_gaslift` or similar?

Comment (joakim-hove) [reply opm/input/eclipse/Schedule/Schedule.hpp:90]: Sure - can use a better name; although the good name for this member is in my opinion something like `gaslift_optimization_is_active`  - which is very long. That undefined ALQ is converted to GRAT (the whole point of the excercise) - is not known/relevant at this particular location.

Comment (joakim-hove) [reply opm/input/eclipse/Schedule/Schedule.hpp:90]: How about `gaslift_opt_active`?

Comment (joakim-hove) [reply opm/input/eclipse/Schedule/Schedule.hpp]: Have changed to `gaslift_opt_active`

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3831 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3833 please

**Comment by hakonhagland** [comment]
@joakim-hove Thanks! This looks good to me.

**Pull request merged**